### ISSUE TITLE
How to automate WPF application which contains chromium web browser?

### ISSUE THREAD

**Issue by FrankieHoomer** [microsoft/WinAppDriver#1068]
I have WPF application, that opens WebPage in ChromiumWebBrowser. Content presenter contains that browser.
How can I get an element from the web page (like button/field/etc)?
I read about context switch from "NATIVE_APP" to "WEBVIEW", but when I try to set AppiumDriver.Context field, it throws NotImplementedException.
Please, any help?

**Comment by anunay1** [comment]
It's not possible.

**Comment by anunay1** [comment]
Yes its possible, using the below code you can switch to an existing browser: 
from selenium import webdriver
from selenium.webdriver.remote.webdriver import WebDriver

# executor_url = driver.command_executor._url
# session_id = driver.session_id

def attach_to_session(executor_url, session_id):
    original_execute = WebDriver.execute
    def new_command_execute(self, command, params=None):
        if command == "newSession":
            # Mock the response
            return {'success': 0, 'value': None, 'sessionId': session_id}
        else:
            return original_execute(self, command, params)
    # Patch the function before creating the driver object
    WebDriver.execute = new_command_execute
    driver = webdriver.Remote(command_executor=executor_url, desired_capabilities={})
    driver.session_id = session_id
    # Replace the patched function with original function
    WebDriver.execute = original_execute
    return driver

bro = attach_to_session('http://127.0.0.1:64092', '8de24f3bfbec01ba0d82a7946df1d1c3')
bro.get('http://ya.ru/')

**Comment by FrankieHoomer** [comment]
Sorry, but looks like it's not C#.
And I don't have the WebDriver type

**Comment by anunay1** [comment]
> Sorry, but looks like it's not C#.
> And I don't have the WebDriver type

Can you check the below link:
https://tarunlalwani.com/post/reusing-existing-browser-session-selenium-csharp/

**Comment by PandaMagnus** [comment]
You can also leverage Chrome's debugging tools to attach to the session as a remote debugger:
https://intellitect.com/selenium-chrome-csharp/
